host/extension.bzl

```diff
@@ -1,3 +1,5 @@
+load("@bazel_skylib//lib:modules.bzl", "modules")
```

Comment:
this would break WORKSPACE users. let's use `mctx.extension_metadata` directly (with `hasattr`)

Reply:
But WORKSPACE users have no reason to load from `extension.bzl`, right? While we could feature detect `mctx.extension_metadata`, that would in turn break Bzlmod users on pre-Bazel 7.1.0 and thus no `reproducible` parameter.

Reply:
WORKSPACE users instantiate a `host_platform_repo` from this file (see local_config_platform.WORKSPACE in Bazel source) We could alternatively maybe separate the repo rule into a separate file, but that's neat in its own way...

Reply:
Ah, thanks for catching that. I submitted a new version that essentially inlines what skylib is doing.